### DISCUSSION TITLE
Diarization Inferrence

### DISCUSSION THREAD

**Discussion by rohaantahir** [speechbrain/speechbrain#1651] [Q&A]
Thanks for the wonderful and excellent library. I have been working on this library for about 6 months and successfully utilize verification and speaker identification modules. Now I want to do diarization as well for that I am facing difficulty in doing inference could you please give me a clue or make some hugging face recipe to utilize diarization module as well so that we only need to input an audio file and it will output the diarization result. Thanks

**Comment by albyho** [comment]
[pyannote-audio](https://github.com/pyannote/pyannote-audio/tree/develop)

``` python
from pyannote.audio.pipelines import SpeakerDiarization
CONFIG = {
    "segmentation": "pyannote/segmentation",
    "embedding": "speechbrain/spkrec-ecapa-voxceleb",
    "clustering": "AgglomerativeClustering"
}
PARAMS = {
    "onset": 0.5, "offset": 0.5,
    "stitch_threshold": 0.04, "min_activity": 6.0,
    "clustering": {"method": "average", "threshold": 0.595}
}

pipeline = SpeakerDiarization(**CONFIG).instantiate(PARAMS)
dia = pipeline("audio.wav")
```

**Comment by mravanelli** [comment]
Thank you @albyho. We still don't have the full diarization pipeline implemented so I think interfacing SpeechBrain with pyannote-audio is the best way to go!

**Comment by rohaantahir** [comment]
@mravanelli any plan to launch end to end diarization testing module in near future?

**Comment by PhantomSpike** [comment]
Hi,

I am also very keen on knowing about the progress status of the diarization pipeline/modules. Do you have even an approximate ETA for this, including maybe some pre-trained models?

If one is to combine SpeechBrain with another diarization-specific solution e.g. pyannote.audio, how exactly would this work? Which parts would you take from pyannote.audio and which ones from SpeechBrain? How would you stitch them together?

Is there anybody who has worked on this and has notebooks/code they are willing to share?

Thanks!
